### ISSUE TITLE
Detrend

### ISSUE THREAD

**Issue by navidcy** [navidcy/LaNina-flavours#1]
Someone was mentioning detrending. Can you help me with detrending? What should I detrend and at which stage.. See notebook at https://github.com/navidcy/LaNina-flavours/blob/main/ACCESS-OM2/access-om2-investigations.ipynb

cc @adisen99, @ClimateFreund, Ruby, @h-hdm

**Comment by adisen99** [comment]
Hey @navidcy 

For detrending, I would suggest using statsmodels tsatools detrending method - [here](https://www.statsmodels.org/stable/generated/statsmodels.tsa.tsatools.detrend.html).

Just vectorise this function using [xarray.apply_ufunc](https://docs.xarray.dev/en/latest/generated/xarray.apply_ufunc.html), for the temperature anomaly data.

I would suggest detrending the temperature anomaly data.


Here is some sample code - 

```python
from statsmodels.tsa.tsatools import detrend

def detrend1d(arr, order=1):
    arr_det = detrend(arr, order) # order=1 for linear detrending
    return arr_det

def detrend3d(da, dim, order=1):
    # da = da.copy()
    return xr.apply_ufunc(detrend1d, da, input_core_dims=[[dim]], output_core_dims=[[dim]], kwargs=dict(order=order), vectorize=True, dask='parallelized')


da_detrended = detrend3d(da, dim='time', order=1)
```

**Comment by navidcy** [comment]
Thanks!

Question: what should I detrend tho? The `temp_anomaly`? Do I detrend just after I remove the climatology and ensure that there is no trend over (i) the whole period or (ii) over the baseline period I computed the climatology from? Or do I detrend after I take the latitudinal average over 5ᵒS-5ᵒN?

**Comment by adisen99** [comment]
> Thanks!
> 
> Question: what should I detrend tho? The `temp_anomaly`? Do I detrend just after I remove the climatology and ensure that there is no trend over (i) the whole period or (ii) over the baseline period I computed the climatology from? Or do I detrend after I take the latitudinal average over 5ᵒS-5ᵒN?

You can detrend the temp_anomaly, and detrend to ensure that there is not trend over the whole period.

**Comment by pancakes175** [comment]
Yes, you can simply detrend the anomaly- if you got it, let us know if there is a trend pls

**Comment by navidcy** [comment]
nice GitHub handle @pancakes175 !

**Comment by navidcy** [comment]
@adisen99 I pushed a new version of the notebook that includes detrending. Could you have a look? If what I did looks good feel free to close this issue.

(I should have pushed into a new branch, made a PR, and tag you... next time ;))

**Comment by adisen99** [comment]
Hey @navidcy, I have taken a look at the code and it looks good to me. I am closing this issue for now.
